Aplib.Core/Metadata.cs

```diff
@@ -27,12 +27,7 @@ public class Metadata
         /// </summary>
         /// <param name="name">The name used to display the component.</param>
         /// <param name="description">The description used to describe the component.</param>
-        public Metadata(string? name = null, string? description = null)
-        {
-            Id = Guid.NewGuid();
-            Name = name;
-            Description = description;
-        }
+        public Metadata(string? name = null, string? description = null) : this(Guid.NewGuid(), name, description) { }
```

Comment:
Is there a specific reason not to do this? Then you don't need the internal constructor. I cannot give a full suggestion for the entire code that would be affected by this, unfortunately, but you get the idea.
```suggestion
        public Metadata(string? name = null, string? description = null)
        {
            Id = Guid.NewGuid();
            Name = name;
            Description = description;
        }
```

Reply:
No the internal constructor was already there. We wanted to be able to give a specific id to some component.